src/analysis/bounds.rs

```diff
@@ -154,12 +160,49 @@ impl Bounds {
                             bound_name,
                         });
                     }
+                } else if par.c_type.ends_with("Func") || par.c_type.ends_with("Callback") ||
+                          par.c_type == "GDestroyNotify" {
```

Comment:
Same here, don't match on names (of types). `GDestroyNotify` should be fine as that's a concrete type, but assuming that all others are ending on `Callback` or `Func` is not going to work well. I know of some instances of `Function` for example, and I'm sure people came up with other things in their libraries.

Reply:
Everything that has a `scope` attribute will be a function pointer for some callback, the `destroy` and `closure`/`user_data` is given by the indexes on the relevant attributes. There should be no need to do guesswork based on type names here.